src/solver/optimisation/opt_rename_problem.h

```diff
@@ -0,0 +1,373 @@
+      = 0;
+    virtual void IntercoIndirectCost(int var,
+                                     const std::string& origin,
+                                     const std::string& destination)
+      = 0;
+    virtual void ShortTermStorageInjection(int var, const std::string& shortTermStorageName) = 0;
+    virtual void ShortTermStorageWithdrawal(int var, const std::string& shortTermStorageName) = 0;
+    virtual void ShortTermStorageLevel(int var, const std::string& shortTermStorageName) = 0;
+    virtual void HydProd(int var) = 0;
+    virtual void HydProdDown(int var) = 0;
+    virtual void HydProdUp(int var) = 0;
+    virtual void Pumping(int var) = 0;
+    virtual void HydroLevel(int var) = 0;
+    virtual void Overflow(int var) = 0;
+    virtual void FinalStorage(int var) = 0;
+    virtual void LayerStorage(int var, int layerIndex) = 0;
+    virtual void PositiveUnsuppliedEnergy(int var) = 0;
+    virtual void NegativeUnsuppliedEnergy(int var) = 0;
+    virtual void AreaBalance(int var) = 0;
+};
+class VariableNamer : public IVariableNamer
+{
+public:
+    using IVariableNamer::IVariableNamer;
+    void DispatchableProduction(int var, const std::string& clusterName) override;
+    void NODU(int var, const std::string& clusterName) override;
+    void NumberStoppingDispatchableUnits(int var, const std::string& clusterName) override;
+    void NumberStartingDispatchableUnits(int var, const std::string& clusterName) override;
+    void NumberBreakingDownDispatchableUnits(int var, const std::string& clusterName) override;
+    void NTCDirect(int var, const std::string& origin, const std::string& destination) override;
+    void IntercoDirectCost(int var,
+                           const std::string& origin,
+                           const std::string& destination) override;
+    void IntercoIndirectCost(int var,
+                             const std::string& origin,
+                             const std::string& destination) override;
+    void ShortTermStorageInjection(int var, const std::string& shortTermStorageName) override;
+    void ShortTermStorageWithdrawal(int var, const std::string& shortTermStorageName) override;
+    void ShortTermStorageLevel(int var, const std::string& shortTermStorageName) override;
+    void HydProd(int var) override;
+    void HydProdDown(int var) override;
+    void HydProdUp(int var) override;
+    void Pumping(int var) override;
+    void HydroLevel(int var) override;
+    void Overflow(int var) override;
+    void FinalStorage(int var) override;
+    void LayerStorage(int var, int layerIndex) override;
+    void PositiveUnsuppliedEnergy(int var) override;
+    void NegativeUnsuppliedEnergy(int var) override;
+    void AreaBalance(int var) override;
+};
+class EmptyVariableNamer : public IVariableNamer
+{
+public:
+    using IVariableNamer::IVariableNamer;
+    void DispatchableProduction(int var, const std::string& clusterName) override
```

Comment:
All these empty functions look like a design problem.
Let's give us time to think about it.

Reply:
A possible way to remove these numerous empty functions would be to encapsulate the **BuildName** function, responsible for the renaming, within a small polymorphic class.
```c++
// opt_rename_problem.h

class BaseNameSender
{
 public:
     void BaseNameSender(std::vector<std::string>& target);
     virtual void putNameAtIndex(int index) = 0;
protected:
     std::vector<std::string>& target_; // Reference to either pb->NomDesVariables or pb->NomDesContraintes
}

class DefaultNameSender : public BaseNameSender
{
using BaseNameSender::BaseNameSender;
public:
    void putNameAtIndex(int index, std::string name) override { target_[index] = name; }
}

class NullNameSender : public BaseNameSender
{
using BaseNameSender::BaseNameSender;
public:
    void putNameAtIndex(int index, std::string name) override { }
}
....

class VariableNamer // No more inheritance
{
public:
    VariableNamer(std::shared_ptr<BaseNameSender> nameSender) : nameSender_(nameSender)
    {}
    ...
private:
    ...
    std::shared_ptr<BaseNameSender> nameSender_;
}
```
```c++
....
void VariableNamer::SetThermalClusterVariableName(int var,
                                                   const std::string& variableType,
                                                   const std::string& clusterName)
{
    const auto location
      = LocationIdentifier(area_, AREA) + SEPARATOR + "ThermalCluster" + "<" + clusterName + ">";

    
    nameSender_->putNameAtIndex(var, BuildName(variableType, location, TimeIdentifier(timeStep_, HOUR));)
}
...

// Here defining a factory for name senders creation.
...
```
```C++
// opt_construction_variables_optimisees_lineaire.cpp
    ...
    auto nameSender = NameSenderFactory(problemeHebdo->NamedProblems);
   std::make_shared<VariableNamer> variableNamer = std::make_shared<VariableNamer>(ProblemeAResoudre);
   ...
  ```